docs/src/service-workers.md

```diff
@@ -0,0 +1,349 @@
+* [`method: Page.route`] and [`method: Page.route`] will **not** see the request (if a Service Worker's fetch handler was registered)
+* [`method: Request.serviceWorker`] will be set to `null`, and [`method: Request.frame`] will return the [Frame]
+* [`method: Response.fromServiceWorker`] will return `true` (if a Service Worker's fetch handler was registered)
+
+Many Service Worker implementations simply execute the request from the page (possibly with some custom caching/offline logic omitted for simplicity):
+
+```js
+// filename: transparent-service-worker.js
+self.addEventListener("fetch", (event) => {
+  // actually make the request
+  const responsePromise = fetch(event.request); 
+  // send it back to the page
+  event.respondWith(responsePromise);
+});
+
+self.addEventListener("activate", (event) => {
+  event.waitUntil(clients.claim());
+});
+```
+
+If a page registers the above Service Worker:
+
+```html
+<!-- filename: index.html -->
+<script>
+  window.registrationPromise = navigator.serviceWorker.register('/transparent-service-worker.js');
+</script>
+```
+
+On the first visit to the page via [`method: Page.goto`], the following Request/Response events would be emitted (along with the corresponding network lifecycle events):
+
+| Event                             | Owner            | URL                            | Routed | [`method: Response.fromServiceWorker`] |
+| -                                 | -                | -                              | -      | -                                      |
+| [`event: BrowserContext.request`] | [Frame]          | index.html                     | Yes    |                                        |
+| [`event: Page.request`]           | [Frame]          | index.html                     | Yes    |                                        |
+| [`event: BrowserContext.request`] | Service [Worker] | transparent-service-worker.js  | Yes    |                                        |
+| [`event: BrowserContext.request`] | Service [Worker] | data.json                      | Yes    |                                        |
+| [`event: BrowserContext.request`] | [Frame]          | data.json                      |        | Yes                                    |
+| [`event: Page.request`]           | [Frame]          | data.json                      |        | Yes                                    |
+
+
+
+It's important to note that, since the example Service Worker just acts a basic transparent "proxy":
+
+* There's 2 [`event: BrowserContext.request`] events for `data.json`; one [Frame]-owned, the other Service [Worker]-owned.
+* Only the Service [Worker]-owned request for the resource was routable via [`method: BrowserContext.route`]; The [Frame]-owned events for `data.json` are not routeable, as they would not have even had the possibility to hit the external network since the Service Worker has a fetch handler registered.
```

Comment:
```suggestion
* Only the Service [Worker]-owned request for the resource was routable via [`method: BrowserContext.route`]; the [Frame]-owned events for `data.json` are not routeable, as they would not have even had the possibility to hit the external network since the Service Worker has a fetch handler registered.
```